internal/oci/mutate/mutate.go

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mutate
+
+import (
+	"errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/sigstore/cosign/internal/oci"
+	"github.com/sigstore/cosign/internal/oci/empty"
+	"github.com/sigstore/cosign/internal/oci/signed"
+)
+
+// Appendable is our signed version of mutate.Appendable
+type Appendable interface {
+	oci.SignedEntity
+
+	MediaType() (types.MediaType, error)
```

Comment:
Is there a reason for this to ever return an error?

Reply:
Rather than explain that this is GGCR copy/pasta I inlined the GGCR interface I copied it from 😉